src/app/blog/frost-threshold-signatures/page.mdx

```diff
@@ -0,0 +1,220 @@
+import { BlogPostLayout } from '@/components/BlogPostLayout'
+import Image from 'next/image'
+
+export const post = {
+  draft: false,
+  author: 'Rüdiger Klaehn',
+  date: '2024-09-30',
+  title: 'Frost threshold signatures',
+  description:
+    'Frost threshold signatures - or how to keep your private keys safe',
+}
+
+export const metadata = {
+  title: post.title,
+  description: post.description,
+}
+
+export default (props) => <BlogPostLayout article={post} {...props} />
+
+## Ed keys everywhere
+
+In [iroh](https://github.com/n0-computer/iroh) we are using [Ed25519] keypairs **a lot**. Nodes are identified by ed keypairs, documents are identified by keypairs, authors, namespaces etc. A gossip topic is an arbitrary 32 byte blob, which conveniently fits an ed public key.
+
+With [pkarr] we have a great mechanism to publish information about keypairs. We are running a dns server, and we can also use the bittorrent [mainline](https://en.wikipedia.org/wiki/Mainline_DHT) DHT for a fully peer to peer mechanism to publish and resolve pkarr packets.
+
+Other recent protocols such as [nostr] are also using ed25519 keypairs.
+
+## How to keep the keys safe
+
+A problem that frequently comes up when using a keypair to control access to an identity or a resource is how to keep the private key safe.
+
+Some keypairs are ephemeral and don't need to be safeguarded much.
+
+Some will have significant security implications from the start (e.g. a keypair associated with access to a crypto wallet).
+
+And some will initially be of low value, but might grow in value over time (e.g. a social media account).
+
+In most cases, there is a constant conflict between the need to keep the keys safe and the need to constantly access the private key for signing messages.
+
+## Existing solutions
+
+### Secure key storage
+
+Most modern hardware supports secure storage for private keys. However, access to such secure storage locations is highly platform dependent. Also, while secure storage makes the key relatively inaccessible, it does not protect against key loss. It also does not provide a mechanism for revocation.
```

Comment:
To add to this: The apple secure enclaves supports P-256 (and I think maybe P-521) only. No Ed25519.
The Android "Keystore" seems to also mostly support ECDSA. I'm not sure how widely EdDSA is supported there either.

Reply:
Oh wow, I did not know that. I thought there were more generic. EdDSA is not exactly new or uncommon. That's dissappointing.

But I don't want this to be an article about the deficiencies of secure storage. So I will just add a sentence that the key formats are limited.